Makefile

```diff
@@ -1,9 +1,8 @@
 SHELL=/bin/bash
 export PATH := ./bin:./venv/bin:$(PATH)
 
-ifndef ELASTIC_VERSION
+# Determine the version to build. Override by setting ELASTIC_VERSION env var.
 ELASTIC_VERSION := $(shell ./bin/elastic-version)
```

Comment:
Sorry, I should have explained the intent better. The idea is to have one, and only one place where the environment variable is checked. With this change, that place is `bin/elastic-version`. It's like poor-engineer's service-oriented architecture, where `bin/elastic-version` is the "Elastic Version Service". It's the one-true-way to determine the ELASTIC_VERSION, even if it has been explicitly specified by the user as an environment variable.

Thus, the Makefile, tests and whatever else don't need to know anything about the environment variable. They just ask and trust the "service", which will always return something valid.